lib/buffer-api.js

```diff
@@ -36,9 +36,14 @@ var util = require('util'),
 
           if (err) {
             return reject(err)
-          } else if (res.statusCode !== 200) {
+          } else if (res.statusCode !== 200 || (body && typeof(body.success) !== "undefined" && !body.success)) {
             if (body) {
-              err = new Error(util.format(body.error))
+              if (body.error) {
```

Comment:
better approach for that, determinate the input source for the method:

```js
let errorMessage
if (body) {
  if (body.error) errorMessage = body.error
  else errorMessage = body.message
}

const error = new Error(util.format(errorMessage))
```

It avoids repeat code